src/rdkafka_cgrp.c

```diff
@@ -2434,6 +2435,272 @@ static void rd_kafka_cgrp_offset_commit_tmr_cb (rd_kafka_timers_t *rkts,
 }
 
 
+/**
+ * @brief Incrementally add to an existing partition assignment
+ *        May update \p partitions but will not hold on to it.
+ *
+ * @returns an error object or NULL on success.
+ */
+static rd_kafka_error_t *
+rd_kafka_cgrp_incremental_assign (rd_kafka_cgrp_t *rkcg,
+                                  rd_kafka_topic_partition_list_t
+                                  *partitions) {
+        int i;
+        rd_kafka_resp_err_t err;
+
+        rd_assert(partitions);
+
+        if (rd_kafka_fatal_error_code(rkcg->rkcg_rk)) {
+                rd_kafka_dbg(rkcg->rkcg_rk, CGRP|RD_KAFKA_DBG_CONSUMER,
+                             "ASSIGN", "Group \"%s\": consumer has raised "
+                             "a fatal error, treating incremental "
+                             "assign of %d partitions as unassign of all"
+                             "%d partition(s) in the current assignment",
+                             rkcg->rkcg_group_id->str, partitions->cnt,
+                             !rkcg->rkcg_assignment ? 0
+                             : rkcg->rkcg_assignment->cnt);
+                err = rd_kafka_cgrp_assign(rkcg, NULL);
+                return rd_kafka_error_new(err, "%s", rd_kafka_err2str(err));
```

Comment:
Why are we using the error code from cgrp_assign() here (will it actually return error?).
The error is that the consumer has raised a fatal error, and that's what this error_new() should say.
So:
`rd_kafka_error_new(RD_KAFKA_RESP_ERR__FATAL, "Consumer has raised a fatal error");`

Reply:
yes it will return an error (a fatal error).
that's why i was using asserts before (which i updated to asserting that the return value was fatal.. before removing it because you were objecting to asserts), to make it clear to the reader.

anyway, your suggestion is objectively best, thanks.